src/LaunchDarkly.Tests/FeatureTest.cs

```diff
@@ -101,26 +101,34 @@ public void IfAUserHasCustomAttributeAsFloat_TargetRulesMatch()
         [Test]
         public void IfAUserHasCustomAttributeAsInteger_TargetRulesMatch()
         {
-            var user = User.WithKey("anyUser").AndCustomAttribute("numbers", 1362823);
             var target = new TargetRule();
-
-            target.Attribute = "numbers";
+            target.Attribute = "Org";
             target.Op = "in";
-            target.Values = new List<object>() { 1362823.0 };
+            target.Values = new List<object>() { 9, 55, 1362823, 292};
 
-            Assert.AreEqual(true, target.Matches(user));
+            var user = User.WithKey("anyUser").AndCustomAttribute("Org", 1362823);
+            Assert.IsTrue(target.Matches(user));
+
+            user = User.WithKey("anyUser").AndCustomAttribute("Org", 55);
+            Assert.IsTrue(target.Matches(user));
+
+            user = User.WithKey("anyUser").AndCustomAttribute("Org", 292);
+            Assert.IsTrue(target.Matches(user));
         }
 
         [Test]
         public void IfAUserHasCustomListAttributeAsIntegers_TargetRulesMatch()
         {
-            var user = User.WithKey("anyUser").AndCustomAttribute("numbers", new List<int>() { 1362823 });
+          
             var target = new TargetRule();
-
-            target.Attribute = "numbers";
+            target.Attribute = "Org";
             target.Op = "in";
             target.Values = new List<object>() { 55, 1362823 };
 
+            var user = User.WithKey("anyUser").AndCustomAttribute("Org", new List<int>() { 1362823 });
```

Comment:
How is it possible for intersection to work but contains fails?



    _____________________________
From: Dan Richelson <notifications@github.com>
Sent: Tuesday, May 17, 2016 7:48 PM
Subject: Re: [launchdarkly/.net-client] Coerce all numbers to doubles before comparison. (#6)
To: launchdarkly/.net-client <".net-client"@noreply.github.com>
Cc: John Kodumal <john@launchdarkly.com>, Comment <comment@noreply.github.com>




In src/LaunchDarkly.Tests/FeatureTest.cs:>              target.Op = "in";>              target.Values = new List<object>() { 55, 1362823 };>  > +            var user = User.WithKey("anyUser").AndCustomAttribute("Org", new List<int>() { 1362823 });

Yes- this exercises the intersection case, which works without coercing the numbers to doubles- as seen here.

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub